### PR TITLE
TPROD-297 Use __serialize() and __unserialize() instead  serialize() and unserialize()

### PR DESCRIPTION
**Pull request by rohitpavaskar**
…alize()

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [n]
| New feature/enhancement? (use the a.x branch)      | [n]
| Deprecations?                          | [y]
| BC breaks? (use the c.x branch)        | [n]
| Automated tests included?              | [n] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | https://mautic.atlassian.net/browse/TPROD-297

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Verify saving plugin credentials and checking those are working fine.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11150"></a>

### CODE REVIEW

File: app/bundles/UserBundle/Security/Authentication/Token/PluginToken.php [hunk 1]
```diff
@@ -85,18 +85,22 @@ public function getResponse()
...
      * {@inheritdoc}
+     *
+     * @param array<mixed> $data
      */
-    public function unserialize($serialized)
+    public function __unserialize(array $data): void
     {
-        list($this->authenticatingService, $this->credentials, $this->providerKey, $parentStr) = unserialize($serialized);
-        parent::unserialize($parentStr);
+        [$this->authenticatingService, $this->credentials, $this->providerKey, $parentStr] = $data;
```
Comment (escopecz) [inline app/bundles/UserBundle/Security/Authentication/Token/PluginToken.php:103]: There is a difference between the original code and the new one. Why did you remove the unserialize function? Shouldn't it be like this?:
```suggestion
        [$this->authenticatingService, $this->credentials, $this->providerKey, $parentStr] = unserialize($data);
```

Comment (rohitp19) [reply app/bundles/UserBundle/Security/Authentication/Token/PluginToken.php:103]: I have made changes according to https://github.com/symfony/security/blob/0a70d429b785affd37c48bb39ddfd38911663a57/Core/Authentication/Token/AbstractToken.php#L191

**Review by RCheesley — APPROVED** [review]
Checked by setting up the Hubspot plugin and all worked as expected - was able to add the API key, save it, and configure the integration.

**Comment by nick-vanpraet** [comment]
Tested by setting up the gmail plugin. Could save the secret, edit it, still there.
Also tested the Twitter plugin, could fill in all the many fields and all values were correctly saved and retrieved upon load.

**Comment by RCheesley** [comment]
Thanks for testing @nick-vanpraet - setting this RTC

**Comment by RCheesley** [comment]
Pretty sure the code review is done already but please just double check @escopecz when you are back :)
